src/content/docs/features/upload.mdx

```diff
@@ -3,12 +3,87 @@ title: Upload
 description: File uploads through HTTP.
 ---
 
-import Stub from '@components/Stub.astro';
 import PluginLinks from '@components/PluginLinks.astro';
+import { Tabs, TabItem, Steps } from '@astrojs/starlight/components';
+import CommandTabs from '@components/CommandTabs.astro';
 
 <PluginLinks plugin="upload" />
 
-<Stub>
-	Based on
-	https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/upload
-</Stub>
+Upload files from disk to a remote server over HTTP. Download files from a remote HTTP server to disk.
+
+## Setup
+
+_This plugin requires a Rust version of at least **1.64**_
+
+<Tabs>
+	<TabItem label="Automatic">
+
+    Use your project's package manager to add the dependency:
+
+    <CommandTabs
+    npm="npm run tauri add upload"
+    yarn="yarn run tauri add upload"
+    pnpm="pnpm tauri add upload"
+    cargo="cargo tauri add upload"
+    />
+
+    </TabItem>
+    <TabItem label="Manual">
+
+    	<Steps>
+
+    	1. Run `cargo add tauri-plugin-upload` to add the plugin to the project's dependencies in `Cargo.toml`.
```

Comment:
I liked the steps, now we should update all the other guides to match this.

Reply:
yeah, I wanted to implement this one so everybody could see them in action. I liked the result tbh.